packages/react/src/components/MultiSelect/next/FilterableMultiSelect.js

```diff
@@ -0,0 +1,598 @@
+    {
+      [`${prefix}--multi-select__wrapper--inline`]: inline,
+      [`${prefix}--list-box__wrapper--inline`]: inline,
+      [`${prefix}--multi-select__wrapper--inline--invalid`]: inline && invalid,
+      [`${prefix}--list-box__wrapper--inline--invalid`]: inline && invalid,
+      [`${prefix}--list-box--up`]: direction === 'top',
+    }
+  );
+  const helperId = !helperText
+    ? undefined
+    : `filterablemultiselect-helper-text-${filterableMultiSelectInstanceId}`;
+  const labelId = `${id}-label`;
+  const titleClasses = cx({
+    [`${prefix}--label`]: true,
+    [`${prefix}--label--disabled`]: disabled,
+    [`${prefix}--visually-hidden`]: hideLabel,
+  });
+  const helperClasses = cx({
+    [`${prefix}--form__helper-text`]: true,
+    [`${prefix}--form__helper-text--disabled`]: disabled,
+  });
+  const inputClasses = cx({
+    [`${prefix}--text-input`]: true,
+    [`${prefix}--text-input--empty`]: !inputValue,
+    [`${prefix}--text-input--light`]: light,
+  });
+  const helper = helperText ? (
+    <div id={helperId} className={helperClasses}>
+      {helperText}
+    </div>
+  ) : null;
+  const menuId = `${id}__menu`;
+  const inputId = `${id}-input`;
+
+  function handleOnChange(changes) {
+    if (onChange) {
+      onChange(changes);
+    }
+  }
+
+  function handleOnMenuChange(forceIsOpen) {
+    const nextIsOpen = forceIsOpen ?? !isOpen;
+    setIsOpen(nextIsOpen);
+    if (onMenuChange) {
+      onMenuChange(nextIsOpen);
+    }
+  }
+
+  function handleOnOuterClick() {
+    handleOnMenuChange(false);
+  }
+
+  function handleOnStateChange(changes, downshift) {
+    if (changes.isOpen && !isOpen) {
+      setTopItems(downshift.selectedItem);
+    }
+
+    const { type } = changes;
+    const { stateChangeTypes } = Downshift;
+
+    switch (type) {
+      case stateChangeTypes.keyDownArrowDown:
+      case stateChangeTypes.keyDownArrowUp:
+      case stateChangeTypes.keyDownHome:
+      case stateChangeTypes.keyDownEnd:
+        setHighlightedIndex(
+          changes.highlightedIndex !== undefined
+            ? changes.highlightedIndex
+            : null
+        );
+        if (stateChangeTypes.keyDownArrowDown === type && !isOpen) {
+          handleOnMenuChange(true);
+        }
+        break;
+      case stateChangeTypes.keyDownEscape:
+        handleOnMenuChange(false);
+        break;
+    }
+  }
+
+  function handleOnInputValueChange(inputValue, { type }) {
+    if (type !== Downshift.stateChangeTypes.changeInput) {
+      return;
+    }
+
+    if (Array.isArray(inputValue)) {
+      setInputValue('');
+    } else {
+      setInputValue(inputValue);
+    }
+
+    if (inputValue && !isOpen) {
+      handleOnMenuChange(true);
+    } else if (!inputValue && isOpen) {
+      handleOnMenuChange(false);
+    }
+  }
+
+  function clearInputValue() {
+    setInputValue('');
+  }
+
+  useEffect(() => {
+    if (textInput.current && inputValue == '') {
+      textInput.current.focus();
+    }
+  }, [inputValue]);
```

Comment:
I think we might get in trouble with this effect based on input value. Is there a place where we are "resetting" the input value that we are trying to hook into so that we can focus the text input? I think we should make sure this happens there instead of in an effect.

This way, the focus will be the result of a user action instead of performed by React when `inputValue` changes.

Reply:
I updated `clearInputValue` to also reset focus to the textInput and removed the effect, let me know if that works!